src/analyses/flow_insensitive_analysis.cpp

```diff
@@ -222,9 +225,11 @@ bool flow_insensitive_analysis_baset::do_function_call(
     t->location_number=1;
 
     locationt l_next=l_call; l_next++;
-    bool new_data=state.transform(ns, l_call, r);
-    new_data = state.transform(ns, r, t) || new_data;
-    new_data = state.transform(ns, t, l_next) || new_data;
+    bool new_data =
+      state.transform(ns, calling_function, l_call, f_it->first, r);
+    new_data = state.transform(ns, f_it->first, r, f_it->first, t) || new_data;
```

Comment:
While you're editing this code, might be sensible to annotate these as "callsite -> head of function" and "callsite -> callsite successor" and so on, explaining the different function names we're passing in

Reply:
Done.